api/gateway-operator/v2alpha1/controlplane_types.go

```diff
@@ -269,6 +275,27 @@ type ControlPlaneK8sCache struct {
 	InitSyncDuration *metav1.Duration `json:"initSyncDuration,omitempty"`
 }
 
+// ControlPlaneConfigDump defines the options for dumping translated Kong configuration from a diagnostics server.
+//
+// +apireference:kgo:include
+// +kubebuilder:validation:XValidation:message="Cannot enable dumpSensitive when state is disabled",rule="self.state == 'enabled' || self.dumpSensitive == 'disabled'"
+type ControlPlaneConfigDump struct {
+	// When State is enabled, Operator will dump the translated Kong configuration by it from a diagnostics server.
+	//
+	// +required
+	// +kubebuilder:validation:Enum=enabled;disabled
+	// +kubebuilder:default="disabled"
+	State ControllerState `json:"state"`
```

Comment:
Can we add a separate type for this instead of reusing an unrelated `ControllerState` type?

Reply:
Used a dedicated type `ConfigDumpState`.